casa/IO/test/tByteSink.cc

```diff
@@ -53,10 +53,10 @@ int main ()
 	uChar    testuChar = 'B';
 	String   testString("This is a teststring");
 
-	RegularFileIO regularFileIO (Path("tByteSink_tmp.dat"),
-				     ByteIO::New);
-	CanonicalIO canonicalIO(&regularFileIO);
-	ByteSink  sink(&canonicalIO);
+        std::shared_ptr<ByteIO> regularFileIO
+          (new RegularFileIO (Path("tByteSink_tmp.dat"), ByteIO::New));
+        std::shared_ptr<TypeIO> canonicalIO (new CanonicalIO(regularFileIO));
```

Comment:
```suggestion
        std::shared_ptr<RegularFileIO> regularFileIO
          (new RegularFileIO (Path("tByteSink_tmp.dat"), ByteIO::New));
        std::shared_ptr<CananicalIO> canonicalIO (new CanonicalIO(regularFileIO));
```
Additionally, you could use `std::make_shared`  here.